lib/puppetserver/ca/action/generate.rb

```diff
@@ -62,19 +83,24 @@ def run(input)
           signer = SigningDigest.new
           return 1 if CliParsing.handle_errors(@logger, signer.errors)
 
+          if input['subject_alt_names'].empty?
+            subject_alt_names = munge_alt_names(puppet.settings[:subject_alt_names])
+          else
+            subject_alt_names = munge_alt_names(input['subject_alt_names'])
+          end
+
           # Generate root and intermediate ca and put all the certificates, crls,
           # and keys where they should go.
-          generate_root_and_intermediate_ca(puppet.settings, signer.digest)
+          generate_pki(puppet.settings, signer.digest, subject_alt_names)
 
           # Puppet's internal CA expects these file to exist.
-          FileSystem.ensure_file(puppet.settings[:serial], "001", 0640)
-          FileSystem.ensure_file(puppet.settings[:cert_inventory], "", 0640)
```

Comment:
Why dont we need to ensure this file anymore? (cert_inventory)?  Was it never needed?

Reply:
Because we're writing it explicitly as part of generating the certs. It needs data from the cert object, so I moved it into the method where those are available.

Reply:
Gotcha